feat(curriculum): add bisection method project to Python (#54110)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d5e3d2927c5e0f4997b.md

---
id: 65ef1d5e3d2927c5e0f4997b
title: Step 17
challengeType: 20
dashedName: step-17
---

# --description--

> Outside the for loop, create an `if` condition to check if the value of `root` is still `None`. If it is, print the message `'Failed to converge within {max_iterations} iterations.'`. Remember to format using an f-string.

Reviewer: ```suggestion
Outside the for loop, create an `if` statement to check if the value of `root` is still `None`. If it is, print the message `'Failed to converge within {max_iterations} iterations.'`. Remember to format the message using an f-string.
```

# --hints--

> You should create an `if` condition that checks `if root is None`.

Reviewer: ```suggestion
You should create an `if` statement that checks if `root is None`.
```

```js
({
    test: () => 
    {
        assert(runPython(`_Node(_code).find_function("square_root_bisection").find_ifs()[1].find_bodies()[2].find_ifs()[0].find_conditions()[0].is_equivalent("root is None")`));

    }
})
```

> You should assign the parameter `f'Failed to converge within {max_iterations} iterations.'` to the `print` function.

Reviewer: ```suggestion
You should pass the argument `f'Failed to converge within {max_iterations} iterations.'` to the `print` function.
```
